### ISSUE TITLE
Offer purchase isn't fully canceled when inventory is full, giving free items

### ISSUE THREAD

**Issue by TroDerp** [TroDerp/EggWars#5]
Reported this myself as it's important to note this issue since it's critic. Video attached reproducing the issue.

I noticed this issue when adding a new message to the merchant shop.

https://github.com/TroDerp/EggWars/assets/72081128/72f5f5c7-48f8-4423-aa43-400379e8bb79

**Comment by TroDerp** [comment]
Hey guys I'm having trouble with a similar thing to this issue; cubecraft's villager shop is now broken and allows you to buy items with the inventory full (and gives you no items), the thing is that I don't know if shift-clicking an offer giving you items till your inventory is full should warn you about it. (making anvil sound even if you did successfully bought same the offer at the same click).
Basically I dont know if it should: Say "Inventory is full" and anvil sound // dont' say anything and make success click sound
